src/request/component-models/LoggingComponentModel.ts

```diff
@@ -44,6 +71,28 @@ export class LogMessageModel implements ILogMessageModel {
         return this._spans;
     }
 
+    private static indexOf(value: string, term: string, window: number) {
+        for (let i = 0; i < value.length && i < window; i++) {
+            if (value[i] === term) {
+                return i;
+            }
+        }
+        
+        return -1;
+    }
+
+    private static lastIndexOf(value: string, term: string, window: number) {
+        const lastWindowIndex = value.length - window;
+        
+        for (let i = value.length - 1; i >= 0 && i >= lastWindowIndex; i--) {
+            if (value[i] === term) {
+                return i;
+            }
+        }
+        
+        return -1;
+    }
+
```

Comment:
Agree that maintaining that state is desirable, but (currently) a component model's lifetime is bound to the component's lifetime (the goal of the model to separate UI logic from business logic for test purposes and not so much to persist state long-term), which means that switching tabs may cause both to be re-created.  (This is one reason why the log level filter state is persisted higher-up, so as to be available as the user switches tabs and/or requests.)

We could change the lifetime of the model, of course, but that also means the model comprehend changes in application (i.e. request) state being pushed down from the parent panels and recognize, for example, when the request has changed (e.g. new messages available) in order to choose to regenerate its state (e.g. ordered/filtered messages and level counts).  Going down that route gets us even further in to `redux` territory and its concept of immutable application state which makes it simpler to identify state changes.  At that point, it may just be better to adopt `redux` instead.